sync_sale_picking_cost/models/sale_order_line.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2025 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    # The cost whis module is syncing is already being computed in this function.
+    # That function has a returned context parameter
+    # that sould be passed if the line is being returned,
+    # if that context is not passed, the purchase price is badly computed
+    @api.depends(
+        "move_ids",
+        "move_ids.stock_valuation_layer_ids",
+        "move_ids.picking_id.state",
+        "qty_delivered",
+    )
+    def _compute_purchase_price(self):
+        res = None
+        for line in self:
+            returned = (
+                any(
+                    line.move_ids.filtered(
+                        lambda m: m.state == "done"
+                    ).origin_returned_move_id
+                )
+                and not line.qty_delivered
```

Comment:
move this comparative first, It is an atomic check and should be much faster to discard the check